src/store/staking/staking.sagas.ts

```diff
@@ -128,13 +131,61 @@ export function* fetchStakesList() {
   }
 }
 
+export function* fetchFilteredHistoryStakeEvents(contractAddresses: string[]) {
+  const subgraphClient = yield* select(subgraphClientSelector);
+
+  if (!subgraphClient) throw new Error('Wallet not connected');
+
+  const { stakeEvents } = yield* call(historyStakesQuery, subgraphClient, {
+    contractAddresses,
+  });
+  return stakeEvents;
+}
+
+export function* getStakesAndVestsAddresses() {
```

Comment:
Actually this saga could be a selector, There is no async logic. It would be easier to test